pkg/config/config_test.go

```diff
@@ -0,0 +1,145 @@
+package config_test
+
+import (
+	"testing"
+
+	"github.com/GoogleCloudPlatform/cloud-run-release-operator/pkg/config"
+	"github.com/stretchr/testify/assert"
+)
+
+var yaml1 = `
+metadata:
+  project: "test"
+  service: "hello"
+  region: "us-east1"
+rollout:
+  steps: [5, 30, 60]
+  interval: 300`
+
+var yaml2 = `
+metadata:
+  service: "hello"
+  region: "us-east1"
+rollout:
+  steps: [5, 30, 60]
+  interval: 300`
+
+var yaml3 = `
+metadata:
+  project: "test"
+  service: "hello"
+  region: "us-east1"
+rollout:
+  interval: 300`
+
+var yaml4 = `
+metadata:
+  project: "test"
+  service: "hello"
+  region: "us-east1"
+rollout:
+  steps: [30, 30, 60]
+  interval: 300`
+
+var yaml5 = `
+metadata:
+  project: "test"
+service: "hello"
+  region: "us-east1"
+  rollout:
+steps: [5, 30, 105]`
+
+var yaml6 = `
+metadata:
+  project: "test"
+  service: "hello"
+  region: "us-east1"
+rollout:
+  steps: [5, 30, 60]`
+
+func TestDecode(t *testing.T) {
+	tests := []struct {
+		name      string
+		data      []byte
+		cliMode   bool
+		expected  *config.Config
+		shouldErr bool
+	}{
+		{
+			name:    "correct config file",
+			data:    []byte(yaml1),
```

Comment:
Any recommendations here? Is it better to put the YAML string in here instead? Would it look good, especially because YAML is very strict about indentation?

Reply:
`string` is ok. but yeah the indentation will be off a bit.